Add gcloud, helm, kubectl to ci image

henrywallace opened the pull request.

kujenga commented on Dockerfile:
```diff
@@ -51,3 +51,15 @@ RUN apt-get update \
+ENV HELM_VERSION 2.3.1
+RUN curl -sS https://storage.googleapis.com/kubernetes-helm/helm-v$HELM_VERSION-linux-amd64.tar.gz -O
+RUN tar -xf helm-v$HELM_VERSION-linux-amd64.tar.gz
+RUN mv linux-amd64/helm /usr/local/bin/helm && rm -rf helm-v$HELM_VERSION-linux-amd64.tar.gz linux-amd64
```
@henrywallace can we combine these helm installation steps into a single command to reduce image size? We should also commit a SHA256 hash and use it to validate the tarball

henrywallace replied:
@kujenga Combining the installation step reduces the image size? Even if I cleanup my unused resources?

kujenga replied:
Yeah, in Docker each step is stored as a separate "layer" with the complete state of the image captured, so if you want to cleanup after yourself, it needs to happen in the same step.

You should be able to mirror the approach taken for Golang above.

dmaze replied:
More specifically, with the three separate `RUN` commands, the `rm` in the third line isn't a net savings of space, since it will be "baked in" to an image layer in the first line.

kujenga commented on Dockerfile:
```diff
@@ -51,3 +51,15 @@ RUN apt-get update \
+RUN echo "deb https://packages.cloud.google.com/apt cloud-sdk-$(lsb_release -sc) main" >> \
+    /etc/apt/sources.list.d/google-cloud-sdk.list
+RUN curl -sS https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+RUN apt-get update && apt-get install -y google-cloud-sdk kubectl
```
Can this line be updated to remove `/var/lib/apt/lists/*` as with the ones above?

kujenga replied:
@henrywallace ping on this concern?

kujenga replied:
also the `--no-install-recommends` flag is worth adding.

dmaze commented:
I don't understand why these steps can't be put in the `circle.yml` file.

I also have somewhat broader concerns over having yet another place that lists out all of the dependencies that needs to be kept up-to-date.  Similar lists of dependencies are in `diffeo-rollout/ansible/roles/development`, `base-images/ubuntu1604-builder`, and the `dev` repo already.

kujenga commented:
@dmaze the idea with a base image is that it speeds up Docker builds by eliminating the need to install things with every single CircleCI build. It's separated out here because at the moment, CircleCI only supports public base images: https://circleci.com/docs/2.0/private-images/

Agreed though about the broader point, but no silver bullet comes to mind. We could look to make whatever components of our base images makes sense open source, and then remove the separation this image creates.

henrywallace commented:
@dmaze The main idea is to speed up CCI's build time. Some of these things steps add a significant amount of time. And while there's an argument for certain CCI build already having long wait times, there are other builds which might migrate to 2.0 which also use the same tools.

And, for now 2.0 doesn't support private containers to run from.

@kujenga I think I've addressed all your concerns.
